### PR TITLE
Add GB4, KP69 and FinnGus Keyboards

### PR DESCRIPTION
**Pull request by JackPikatea**
## Description

Add KP69 Keyboard. Add GB4 Macropad. Add FinnGus keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pikatea/finngus/finngus.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "finngus.h"
+
+#if defined(ENCODER_ENABLE)
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/pikatea/finngus/finngus.c:18]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
        return false;
    }
```

File: keyboards/pikatea/finngus/readme.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# pikatea/finngus
+
+![pikatea](https://i.imgur.com/tEjK2Mr.png)
```
Comment (drashna) [inline keyboards/pikatea/finngus/readme.md:3]:
```suggestion

![pikatea](https://i.imgur.com/tEjK2Mrh.png)
```

File: keyboards/pikatea/finngus/rules.mk [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/pikatea/finngus/rules.mk:18]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/pikatea/finngus/rules.mk [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/pikatea/finngus/rules.mk:21]:
```suggestion
```

File: keyboards/pikatea/finngus/rules.mk [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/pikatea/finngus/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
```

File: keyboards/pikatea/pikatea_keyboard_kp69/keymaps/midi/config.h [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline keyboards/pikatea/pikatea_keyboard_kp69/keymaps/midi/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/pikatea/pikatea_keyboard_kp69/keymaps/midi_encoder/config.h [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline keyboards/pikatea/pikatea_keyboard_kp69/keymaps/midi_encoder/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/pikatea/pikatea_keyboard_kp69/keymaps/midi_encoder/keymap.c [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+};
+
+//create a virtual Midi Slider with the encoder
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+        midi_send_cc(&midi_device, 0, currentLayer+1, 65);
+    } else {
+        midi_send_cc(&midi_device, 0, currentLayer+1, 63);
+    }
+    return true;
```
Comment (drashna) [inline keyboards/pikatea/pikatea_keyboard_kp69/keymaps/midi_encoder/keymap.c:68]:
```suggestion
    return false;
```

File: keyboards/pikatea/pikatea_keyboard_kp69/pikatea_keyboard_kp69.c [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "pikatea_keyboard_kp69.h"
+
+#if !defined(VIA_ENABLE) && !defined(MIDI_ENABLE) && defined(ENCODER_ENABLE)
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/pikatea/pikatea_keyboard_kp69/pikatea_keyboard_kp69.c:20]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
```

File: keyboards/pikatea/pikatea_keyboard_kp69/rules.mk [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/pikatea/pikatea_keyboard_kp69/rules.mk:18]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/pikatea/pikatea_keyboard_kp69/rules.mk [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/pikatea/pikatea_keyboard_kp69/rules.mk:21]:
```suggestion
```

File: keyboards/pikatea/pikatea_macropad_gb4/config.h [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+/*== or choose animations ==*/
+// #    define RGBLIGHT_EFFECT_BREATHING
+// #    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+// #    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+// #    define RGBLIGHT_EFFECT_SNAKE
+// #    define RGBLIGHT_EFFECT_KNIGHT
+// #    define RGBLIGHT_EFFECT_CHRISTMAS
+// #    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+// #    define RGBLIGHT_EFFECT_RGB_TEST
+// #    define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/pikatea/pikatea_macropad_gb4/config.h:80]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/pikatea/pikatea_macropad_gb4/keymaps/midi/config.h [hunk 13]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline keyboards/pikatea/pikatea_macropad_gb4/keymaps/midi/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/pikatea/pikatea_macropad_gb4/readme.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+![pikatea](https://i.imgur.com/UpcSHLi.png)
+
+*3, 5 or 7 keys and a knob mounted to an underside surface*
+
+* Keyboard Maintainer: [Jack Kester](https://github.com/JackPikatea)
+* Hardware Supported: *M32u4*
+* Hardware Availability: *http://www.pikatea.com*
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/pikatea/pikatea_macropad_gb4/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/pikatea/pikatea_macropad_gb4/readme.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
+# pikatea/pikatea_macropad_gb4
+
+![pikatea](https://i.imgur.com/UpcSHLi.png)
```
Comment (drashna) [inline keyboards/pikatea/pikatea_macropad_gb4/readme.md:3]:
```suggestion
![pikatea](https://i.imgur.com/UpcSHLih.png)
```

File: keyboards/pikatea/pikatea_macropad_gb4/rules.mk [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/pikatea/pikatea_macropad_gb4/rules.mk:18]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/pikatea/pikatea_macropad_gb4/rules.mk [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/pikatea/pikatea_macropad_gb4/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by JackPikatea** [comment]
@drashna Thank you for reviewing. I've made the changes you suggested.

**Comment by drashna** [comment]
You still have a number of outstanding review comments that need to be addressed.  Copyright headers, and more. 

Double check the "hidden conversations", as I think you missed those, specifically.

**Pull request closed without merging**